### DISCUSSION TITLE
Inertia router.post generates error "All Inertia requests must receive a valid Inertia response, however a plain JSON response was received."

### DISCUSSION THREAD

**Discussion by simonjcarr** [adonisjs/core#4734] [General]
Hope someone can help, this driving me nuts and I know it should be so easy, so I am obviously doing something wrong.

In my react component I have

```
import { Head, router, usePage } from "@inertiajs/react";
...
console.log(values)
router.post('/register', values);
...
```
`values` has the following value 
```
{
    "name": "sdfsd",
    "email": "sdfsdf@sdsdf.com",
    "password": "password",
    "password_confirmation": "password"
}
```

I have the following route
```
router.post('/register', [RegisterController, 'store']).as('register.store')
```

I have the following controller method

```
  public async store({ request, response }: HttpContext) { 
        const data = request.all()
        console.log(data);
        response.redirect('/')
    }
```
`data` = `sdfsd sdfsdf@sdsdf.com password`

I don't get redirected to `/`, instead I stay on the `/register` page and the following modal pops up

![image](https://github.com/user-attachments/assets/d856a06f-d265-4d03-afe5-4f21b61f11b7)

Any ideas what I am doing wrong here?

**Comment by simonjcarr** [comment]
It was as simple as restarting adonis. Not sure if the problem is because I am using WSL, but the frontend is updating the backend wasn't

**Reply by RomainLanz** [reply]
Are you lazy-importing your controller as showed in the documentation?

It is needed with HMR.

📚 https://docs.adonisjs.com/guides/concepts/hot-module-replacement#works-only-with-dynamic-imports
